"error: out of range pc-relative fixup value" on linux-armel checked CoreCLR_NonPortable

linux-armel checked CoreCLR_NonPortable:

```
  [ 37%] Building ASM object vm/wks/CMakeFiles/cee_wks_core.dir/__/arm/thunktemplates.S.o
  /tmp/thunktemplates-147a73.s:348:9: error: out of range pc-relative fixup value
          ldr pc, StubPrecodeCode + PAGE_SIZE + 0x04
          ^
  /tmp/thunktemplates-147a73.s:355:9: error: out of range pc-relative fixup value
          ldr r12, FixupPrecodeCode + PAGE_SIZE + 0x04
          ^
  /tmp/thunktemplates-147a73.s:356:9: error: out of range pc-relative fixup value
          ldr pc, FixupPrecodeCode + PAGE_SIZE + 0x08
          ^
  /tmp/thunktemplates-147a73.s:369:9: error: out of range pc-relative fixup value
          ldr pc, CallCountingStubCode + PAGE_SIZE + 0x04
          ^
  /tmp/thunktemplates-147a73.s:371:9: error: out of range pc-relative fixup value
          ldr pc, CallCountingStubCode + PAGE_SIZE + 0x08
          ^
  make[2]: *** [vm/wks/CMakeFiles/cee_wks_core.dir/build.make:3470: vm/wks/CMakeFiles/cee_wks_core.dir/__/arm/thunktemplates.S.o] Error 1
  make[1]: *** [CMakeFiles/Makefile2:3712: vm/wks/CMakeFiles/cee_wks_core.dir/all] Error 2
```

E.g.:
https://dev.azure.com/dnceng-public/public/_build/results?buildId=610702&view=logs&j=2eda3965-9146-55d1-4226-c3d3f8432d8f&t=27bbe0a6-6c05-550a-bc6f-305cf5eac8aa
<!-- Error message template  -->
### Known Issue Error Message

Fill the error message using [step by step known issues guidance](https://github.com/dotnet/arcade/blob/main/Documentation/Projects/Build%20Analysis/KnownIssues.md#how-to-fill-out-a-known-issue-error-section).

<!-- Use ErrorMessage for String.Contains matches. Use ErrorPattern for regex matches (single line/no backtracking). Set BuildRetry to `true` to retry builds with this error. Set ExcludeConsoleLog to `true` to skip helix logs analysis. -->

```json
{
  "ErrorMessage": "out of range pc-relative fixup value",
  "ErrorPattern": "",
  "BuildRetry": false,
  "ExcludeConsoleLog": false
}
```


<!-- Known issue validation start -->
 ### Known issue validation
**Build: :mag_right:** https://dev.azure.com/dnceng-public/public/_build/results?buildId=610702
**Error message validated:** `[out of range pc-relative fixup value`]
**Result validation:** :white_check_mark: Known issue matched with the provided build.
**Validation performed at:** 3/20/2024 10:47:26 PM UTC
<!-- Known issue validation end -->
<!--Known issue error report start -->
### Report
|Build|Definition|Step Name|Console log|Pull Request|
|---|---|---|---|---|
|[611731](https://dev.azure.com/dnceng-public/public/_build/results?buildId=611731)|dotnet/runtime|Build product|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/611731/logs/390)|dotnet/runtime#100076|
|[611594](https://dev.azure.com/dnceng-public/public/_build/results?buildId=611594)|dotnet/runtime|Build product|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/611594/logs/424)|dotnet/runtime#100012|
|[611588](https://dev.azure.com/dnceng-public/public/_build/results?buildId=611588)|dotnet/runtime|Build product|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/611588/logs/355)|dotnet/runtime#100011|
|[611340](https://dev.azure.com/dnceng-public/public/_build/results?buildId=611340)|dotnet/runtime|Build product|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/611340/logs/2012)|dotnet/runtime#99836|
|[611560](https://dev.azure.com/dnceng-public/public/_build/results?buildId=611560)|dotnet/runtime|Build product|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/611560/logs/512)|dotnet/runtime#99953|
|[611533](https://dev.azure.com/dnceng-public/public/_build/results?buildId=611533)|dotnet/runtime|Build product|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/611533/logs/271)|dotnet/runtime#100018|
|[611406](https://dev.azure.com/dnceng-public/public/_build/results?buildId=611406)|dotnet/runtime|Build product|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/611406/logs/244)|dotnet/runtime#99608|
|[611393](https://dev.azure.com/dnceng-public/public/_build/results?buildId=611393)|dotnet/runtime|Build product|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/611393/logs/539)|dotnet/runtime#99956|
|[611371](https://dev.azure.com/dnceng-public/public/_build/results?buildId=611371)|dotnet/runtime|Build product|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/611371/logs/211)|dotnet/runtime#100021|
|[611360](https://dev.azure.com/dnceng-public/public/_build/results?buildId=611360)|dotnet/runtime|Build product|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/611360/logs/680)||
|[611333](https://dev.azure.com/dnceng-public/public/_build/results?buildId=611333)|dotnet/runtime|Build product|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/611333/logs/536)|dotnet/runtime#96169|
|[611323](https://dev.azure.com/dnceng-public/public/_build/results?buildId=611323)|dotnet/runtime|Build product|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/611323/logs/528)|dotnet/runtime#96332|
|[611290](https://dev.azure.com/dnceng-public/public/_build/results?buildId=611290)|dotnet/runtime|Build product|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/611290/logs/247)|dotnet/runtime#100000|
|[611265](https://dev.azure.com/dnceng-public/public/_build/results?buildId=611265)|dotnet/runtime|Build product|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/611265/logs/271)|dotnet/runtime#94250|
|[611256](https://dev.azure.com/dnceng-public/public/_build/results?buildId=611256)|dotnet/runtime|Build product|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/611256/logs/245)|dotnet/runtime#100060|
|[611160](https://dev.azure.com/dnceng-public/public/_build/results?buildId=611160)|dotnet/runtime|Build product|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/611160/logs/226)|dotnet/runtime#99778|
|[611096](https://dev.azure.com/dnceng-public/public/_build/results?buildId=611096)|dotnet/runtime|Build product|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/611096/logs/259)|dotnet/runtime#97529|
|[611080](https://dev.azure.com/dnceng-public/public/_build/results?buildId=611080)|dotnet/runtime|Build product|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/611080/logs/246)|dotnet/runtime#100054|
|[611058](https://dev.azure.com/dnceng-public/public/_build/results?buildId=611058)|dotnet/runtime|Build product|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/611058/logs/237)|dotnet/runtime#98258|
|[611008](https://dev.azure.com/dnceng-public/public/_build/results?buildId=611008)|dotnet/runtime|Build product|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/611008/logs/252)|dotnet/runtime#100050|
|[610999](https://dev.azure.com/dnceng-public/public/_build/results?buildId=610999)|dotnet/runtime|Build product|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/610999/logs/226)|dotnet/runtime#99991|
|[610702](https://dev.azure.com/dnceng-public/public/_build/results?buildId=610702)|dotnet/runtime|Build product|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/610702/logs/489)|dotnet/runtime#99818|
|[610676](https://dev.azure.com/dnceng-public/public/_build/results?buildId=610676)|dotnet/runtime|Build product|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/610676/logs/200)|dotnet/runtime#98258|
|[610632](https://dev.azure.com/dnceng-public/public/_build/results?buildId=610632)|dotnet/runtime|Build product|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/610632/logs/208)|dotnet/runtime#100041|
#### Summary
|24-Hour Hit Count|7-Day Hit Count|1-Month Count|
|---|---|---|
|24|24|24|
<!--Known issue error report end -->

cc @filipnavara, clang was updated yesterday from v17 to v18, which broke the armel build. Is this our bug or compiler's?

```sh
  -- The C compiler identification is Clang 18.1.2
  -- The CXX compiler identification is Clang 18.1.2
...
  [ 37%] Building ASM object nativeaot/Runtime/Full/CMakeFiles/Runtime.ServerGC.dir/__/arm/Interlocked.S.o
  /tmp/thunktemplates-9c0dbb.s:348:9: error: out of range pc-relative fixup value
          ldr pc, StubPrecodeCode + PAGE_SIZE + 0x04
          ^
  /tmp/thunktemplates-9c0dbb.s:355:9: error: out of range pc-relative fixup value
          ldr r12, FixupPrecodeCode + PAGE_SIZE + 0x04
          ^
  /tmp/thunktemplates-9c0dbb.s:356:9: error: out of range pc-relative fixup value
          ldr pc, FixupPrecodeCode + PAGE_SIZE + 0x08
          ^
  /tmp/thunktemplates-9c0dbb.s:369:9: error: out of range pc-relative fixup value
          ldr pc, CallCountingStubCode + PAGE_SIZE + 0x04
          ^
  /tmp/thunktemplates-9c0dbb.s:371:9: error: out of range pc-relative fixup value
          ldr pc, CallCountingStubCode + PAGE_SIZE + 0x08
          ^
  make[2]: *** [vm/wks/CMakeFiles/cee_wks_core.dir/build.make:3470: vm/wks/CMakeFiles/cee_wks_core.dir/__/arm/thunktemplates.S.o] Error 1
  make[1]: *** [CMakeFiles/Makefile2:3712: vm/wks/CMakeFiles/cee_wks_core.dir/all] Error 2
  make[1]: *** Waiting for unfinished jobs....
```

--- a/src/coreclr/vm/arm/thunktemplates.S
+++ b/src/coreclr/vm/arm/thunktemplates.S
@@ -11,7 +11,7 @@
 
 PAGE_SIZE = 4096
 
-#define DATA_SLOT(stub, field) stub##Code + PAGE_SIZE + stub##Data__##field
+#define DATA_SLOT(stub, field) . - (. - stub##Code) + PAGE_SIZE + stub##Data__##field
 
     LEAF_ENTRY StubPrecodeCode
         ldr r12, DATA_SLOT(StubPrecode, MethodDesc)

